### ISSUE TITLE
Running multiple Azure steps in parallel causes file locking issue when unpacking Calamari

### ISSUE THREAD

**Issue by alfhenrik** [OctopusDeploy/Issues#2079]
This only happens the first time you run an Azure step as we haven't unpacked Calamari yet, any subsequent deployments will work fine. We should implement similar logic as we do on Tentacles to not push the package more than 1 at a time.

![image](https://cloud.githubusercontent.com/assets/1035315/10501894/5b186bfc-7329-11e5-9ea0-b7c7c5e5fae4.png)

**Comment by alfhenrik** [comment]
Duplicate of #1998
